vault/acl.go

```diff
@@ -47,25 +50,58 @@ func NewACL(policies []*Policy) (*ACL, error) {
 			// Check for an existing policy
 			raw, ok := tree.Get(pc.Prefix)
 			if !ok {
-				tree.Insert(pc.Prefix, pc.CapabilitiesBitmap)
+				tree.Insert(pc.Prefix, pc.Permissions)
 				continue
 			}
-			existing := raw.(uint32)
+
+			// these are the ones already in the tree
+			existingPerms := raw.(*Permissions)
```

Comment:
During an upgrade, I feel that this can panic. How about we try to cast the type, if it fails, cast this to uint32 and have an upgrade path?